### PR TITLE
Improve CI configuration

### PR DESCRIPTION
**Pull request by swissspidy**
* Fix definition of global env variable
* Remove incorrect docker volumes
* Stop local environment after running e2e tests
* Add demo plugin that demonstrates how plugins can be used during e2e tests
* Add new CI step to run e2e tests as author

Related: #2795.

### CODE REVIEW

**Comment by westonruter** [comment]
Why is the demo plugin needed here?

**Comment by swissspidy** [comment]
@westonruter It basically acts as documentation to show how adding plugins for E2E tests works.

Alternatively we can just wait until we actually need some of these plugins when adding the first bunch of tests.

**Comment by westonruter** [comment]
I'm confused though: these tests are for the AMP plugin. Why would adding plugins need to be tested in this project?

**Comment by westonruter** [comment]
Unless the intention is to test that AMP validation errors are getting properly attributed?

**Comment by swissspidy** [comment]
@westonruter It's not about testing plugins. This change allows for adding some utility plugins that can be activated during E2E tests. 

For example, such a plugin could create a custom post type and the accompanying E2E test would verify that block validation errors are shown for this custom post type too.

File: tests/e2e/plugins/demo.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+/**
+ * Plugin Name: E2E Tests Demo Plugin
+ * Plugin URI:  https://github.com/ampproject/amp-wp
+ * Description: Demo Plugin that can be installed during E2E tests.
```
Comment (miina) [inline tests/e2e/plugins/demo.php]: > For example, such a plugin could create a custom post type and the accompanying E2E test would verify that block validation errors are shown for this custom post type too.

@swissspidy Could we add this clarifying example to the description here, too?

Comment (swissspidy) [reply tests/e2e/plugins/demo.php]: @miina Sure. Maybe in `contributing.md` though?

Comment (miina) [reply tests/e2e/plugins/demo.php]: Yep, makes sense 👍

Comment (swissspidy) [reply tests/e2e/plugins/demo.php]: Done in 2fdebe1ed017294154c015864e656c4a94b9f037

**Pull request merged**